service/src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java

```diff
@@ -349,11 +349,10 @@ public String getRequiredGcpProject(UUID workspaceId) {
   public String enablePetServiceAccountImpersonation(
       UUID workspaceId, AuthenticatedUserRequest userRequest) {
     final String serviceAccountUserRole = "roles/iam.serviceAccountUser";
-    // Validate that the user is at least a writer in this workspace, as only writers can run
-    // workflows.
+    // Validate that the user is a member of the workspace.
     Workspace workspace =
         validateWorkspaceAndAction(
-            userRequest, workspaceId, SamConstants.SAM_WORKSPACE_WRITE_ACTION);
+            userRequest, workspaceId, SamConstants.SAM_WORKSPACE_READ_ACTION);
```

Comment:
What's the reason for this change? If it's just to support the CLI, then we could also handle it in the CLI (e.g. check whether a user is a reader or writer on the workspace before calling the endpoint, or just tolerating failures there, etc.). It feels weird that a reader can modify resources in the workspace project, though I suppose they can already always create a pet SA via SAM regardless of their role in the workspace.

Reply:
This is mainly to support the CLI, but I also don't think there's a particularly strong case for this to be WRITER instead of READER. Pet impersonation is required for pipelines API jobs, but the user also needs PAPI specific permissions that only writers have for that. By itself, pet impersonation doesn't really allow users to do anything.
We also don't revoke this when a user's roles change, so even with the current system someone could do:
1. Add user X as a writer
2. Enable user X's pet SA
3. Add user X as a reader
4. Remove user X as a writer

and end up in the same place.

Reply:
Good points. How about keeping this as READER then, instead of tying it to WRITER/permission to run PAPI jobs? I think the CLI wouldn't need to change -- we still call enablePet each time we fetch the key file.

Reply:
sounds good, this check will be READER and the pending CLI changes should work as-is.